File: app/src/main/java/tomaszkopacz/meetbam/service/LoginService.java
package tomaszkopacz.meetbam.service;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;
/**
 * Created by tomas on 05.04.2018.
 */

public class LoginService {

    private static SharedPreferences sharedPreferences;

    private static final String MAIL = "mail";
    private static final String NAME = "name";
    private static final String SURNAME = "surname";
    private static final String PASSWORD = "password";
    private static final String NO_USER = "";

    public LoginService(Context context){

        this.sharedPreferences = PreferenceManager
                .getDefaultSharedPreferences(context);
    }

    public boolean isUserLoggedIn(){

        String user = sharedPreferences.getString(MAIL, NO_USER);

        return (!user.equals(NO_USER)) ? true : false;
    }

    public static String getUserMail(){
        return sharedPreferences.getString(MAIL, NO_USER);
    }

    public static String getUserName(){
        return sharedPreferences.getString(NAME, NO_USER);
    }

    public static String getUserSurname(){
        return sharedPreferences.getString(SURNAME, NO_USER);
    }

    public static String getUserPassword(){
        return sharedPreferences.getString(PASSWORD, NO_USER);
    }


    public void login(String mail, String name, String surname, String pwd){
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putString(MAIL, mail);
        editor.putString(NAME, name);
        editor.putString(SURNAME, surname);
        editor.putString(PASSWORD, pwd);
        editor.commit();
    }

    public void logout(){
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.remove(MAIL);
        editor.remove(NAME);
        editor.remove(SURNAME);
        editor.remove(PASSWORD);
        editor.commit();
    }
}

File: app/src/main/java/tomaszkopacz/meetbam/service/LoginService.kt
package tomaszkopacz.meetbam.service

import android.content.Context
import tomaszkopacz.meetbam.entity.User

class LoginService(context: Context) {

    private var prefs: Preferences = Preferences(context)

    fun login(mail: String, name: String, surname: String, password: String) {

        prefs.name = name
        prefs.surname = surname
        prefs.mail = mail
        prefs.password = password
    }

    fun logout() {
        prefs.removeAll()
    }

    fun isUserLoggedIn(): Boolean {
        return (prefs.name != prefs.NO_USER && prefs.surname != prefs.NO_USER &&
                prefs.mail != prefs.NO_USER && prefs.password != prefs.NO_USER)
    }

    fun getLoggedUser(): User{
        var user = User()

        if (isUserLoggedIn()) {
            user.name = prefs.name
            user.surname = prefs.surname
            user.mail = prefs.mail
            user.password = prefs.password
        }

        return user
    }
}
